include/FreeRTOSIPConfigDefaults.h

```diff
@@ -652,4 +652,8 @@
     #define ipconfigPROCESS_CUSTOM_ETHERNET_FRAMES    0
 #endif
 
+#ifnef ipconfigDONOT_COPY_SOCKETSET_TO_CHIELD_SOCKET
+	#define ipconfigDONOT_COPY_SOCKETSET_TO_CHIELD_SOCKET	0
```

Comment:
I assume that the macro contains a typo? `CHIELD` should be `CHILD`?
Or maybe the macro can be rephrased as `CHILD_SOCKETS_INHERIT_SOCKET_SET`, by default defined as `1`?
( positive statements are easier to understand than macros containing "NOT" or "DONT" ).

**About inheriting**: sockets that are obtained by calling `accept()` already exists for a short while. It is possible that the socket already has received data, and `recv()` may be called. It also means that the socket has created a reception stream buffer. The size of the buffer is also `inherited` from the parent socket.
Now when a socket receives data, I want to have the socket set triggered. That is the reason why it is copied from the parent.
But as @AniruddhaKanhere suggests, you can also attach the socket to a different socket set right after obtaining it with `FreeRTOS_accept()`:
~~~c
Socket_t xSocket = FreeRTOS_accept( xServerSocket, &( xAddress ), &( xAddressLength ) );

if( xSocketValid( xSocket ) )
{
    FreeRTOS_FD_SET( xSocket,
                     xSocketSet,
                     eSELECT_EXCEPT | eSELECT_READ );
}
~~~